tsfresh/feature_selection/relevance.py

```diff
@@ -0,0 +1,254 @@
+                      features among all created features.
+    :type fdr_level: float
+
+    :param hypotheses_independent: Can the significance of the features be assumed to be independent?
+                                   Normally, this should be set to False as the features are never
+                                   independent (e.g. mean and median)
+    :type hypotheses_independent: bool
+
+    :param n_jobs: Number of processes to use during the p-value calculation
+    :type n_jobs: int
+
+    :param chunksize: Size of the chunks submitted to the worker processes
+    :type chunksize: int
+
+    :return: A pandas.DataFrame with each column of the input DataFrame X as index with information on the significance
+             of this particular feature. The DataFrame has the columns
+             "Feature",
+             "type" (binary, real or const),
+             "p_value" (the significance of this feature as a p-value, lower means more significant)
+             "relevant" (True if the Benjamini Hochberg procedure rejected the null hypothesis [the feature is
+             not relevant] for this feature)
+    :rtype: pandas.DataFrame
+    """
+    if ml_task not in ['auto', 'classification', 'regression']:
+        raise ValueError('ml_task must be one of: \'auto\', \'classification\', \'regression\'')
+    elif ml_task == 'auto':
+        ml_task = infer_ml_task(y)
+
+    if n_jobs == 0:
+        map_function = map
+    else:
+        pool = Pool(n_jobs)
+        map_function = partial(pool.map, chunksize=chunksize)
+
+    relevance_table = pd.DataFrame(index=pd.Series(X.columns, name='feature'))
+    relevance_table['feature'] = relevance_table.index
+    relevance_table['type'] = pd.Series(
+        map_function(get_feature_type, [X[feature] for feature in relevance_table.index]),
+        index=relevance_table.index
+    )
+    table_real = relevance_table[relevance_table.type == 'real'].copy()
+    table_binary = relevance_table[relevance_table.type == 'binary'].copy()
+
+    table_const = relevance_table[relevance_table.type == 'constant'].copy()
+    table_const['p_value'] = np.NaN
+    table_const['relevant'] = False
+
+    if len(table_const) == len(relevance_table):
+        return table_const
+
+    if ml_task == 'classification':
+        tables = []
+        for label in y.unique():
+            tmp = _calculate_relevance_table_for_binary_target(
+                table_real, table_binary, X, (y == label),
+                test_for_binary_target_real_feature=test_for_binary_target_real_feature,
+                hypotheses_independent=hypotheses_independent, fdr_level=fdr_level, map_function=map_function
+            )
+            tables.append(tmp)
+        relevance_table = combine_relevance_tables(tables)
+    elif ml_task == 'regression':
+        _calculate_real_feature = partial(target_real_feature_real_test, y=y)
```

Comment:
I'd have created a more general function that would fit both cases, taking the significance tests as parameters. But as the `target_binary_feature_real_test` has an additional parameter I wanted to avoid the complexity of checking for that case.
However, reviewing it, I see that I could just pass in the significance tests as configured partials. Then I can remove the duplication.